### PR TITLE
Glooctl upstreamgroups get, create & delete commands

### PR DESCRIPTION
**Pull request by kdorosh**
Adds glooctl support for upstream groups, get/create/delete commands.

Opted to have glooctl only support upstream weighted destinations during creation because:
- the user experience for providing kubernetes service destinations and consul destinations would look hairy
- it's simpler to create the kubernetes service/consule destinations in upstream objects and then link them into upstream groups
- `Upstream` supports more types of destinations than the `WeightedDestination`s object does at the highest level anyways, and it's easier to bundle those types in upstreams first

Opened issues for:
- glooctl edit command for upstream groups https://github.com/solo-io/gloo/issues/894
- refactoring duplicated resource collect logic in `gloo/cli/pkg/surveryutils` https://github.com/solo-io/gloo/issues/895
  - this PR slightly contributes to the problem here, but the problem is far larger and worthy of its own issue & PR

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/649

### CODE REVIEW

File: projects/gloo/cli/pkg/cmd/create/upstreamgroup.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+func createUpstreamGroup(opts *options.Options) error {
+	ug, err := upstreamGroupFromOpts(opts)
+	if err != nil {
+		return err
+	}
+
+	if opts.Create.DryRun {
+		return common.PrintKubeCrd(ug, v1.UpstreamGroupCrd)
+	}
```
Comment (yuval-k) [inline projects/gloo/cli/pkg/cmd/create/upstreamgroup.go]: and ability to print gloo yamls was added here: https://github.com/solo-io/gloo/pull/897
this should be updated to inlcude similar logic

File: projects/gloo/cli/pkg/cmd/create/upstreamgroup_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	Context("Invalid format", func() {
+		It("should error when namespaced upstreams have invalid format", func() {
+			err := testutils.Glooctl("create upstreamgroup test --namespace gloo-system --weighted-upstreams us1=2,us2=3")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("invalid format: provide namespaced upstream names"))
+		})
+	})
+
+	Context("It works", func() {
+		expectedDest := []*v1.WeightedDestination{
```
Comment (yuval-k) [inline projects/gloo/cli/pkg/cmd/create/upstreamgroup_test.go]: it best practice to initialize variables inside a BeforeEach:
```
var (
foo string
)
BeforeEach(func(){foo="bar"})
```

this has the advantage of make sure that even if one test changes the variable it will be cleaned for the next test

File: projects/gloo/cli/pkg/printers/upstreamgroup.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+		weight := fmt.Sprint(totalWeight(ug))
+		details := upstreamGroupDetails(ug)
+
+		if len(details) == 0 {
+			details = []string{""}
+		}
+		for i, line := range details {
+			if i == 0 {
+				table.Append([]string{name, status, weight, line})
```
Comment (yuval-k) [inline projects/gloo/cli/pkg/printers/upstreamgroup.go]: IIUC, the total weight will be displayed next to the first upstream in the group. This might be confusing. IMHO we should put the name,status,weight on their own line, followed by lines with the details.

File: projects/gloo/cli/pkg/surveyutils/route.go [hunk 4]
```diff
@@ -263,11 +262,10 @@ func AddRouteFlagsInteractive(opts *options.Options) error {
 		}
 	} else {
 		// only get the insert index if the vs is predefined
-		if err := cliutil.GetUint32InputDefault(
+		if err := cliutil.GetUint32Input(
```
Comment (yuval-k) [inline projects/gloo/cli/pkg/surveyutils/route.go]: why this change?

Comment (kdorosh) [reply projects/gloo/cli/pkg/surveyutils/route.go]: its just a functionally equivalent, shorter version of the old method: 
https://github.com/solo-io/gloo/blob/master/pkg/cliutil/input.go#L52

new method: https://github.com/solo-io/gloo/blob/master/pkg/cliutil/input.go#L48

the change is just for brevity

File: projects/gloo/cli/pkg/printers/upstreamgroup.go (hunk 3 above)
Comment (kdorosh) [reply projects/gloo/cli/pkg/printers/upstreamgroup.go]: It renders like so. The total weight has its own column, so I don't think its confusing. Open to suggestions here
```
+----------------+----------+--------------+--------------------------------+
| UPSTREAM GROUP |  STATUS  | TOTAL WEIGHT |            DETAILS             |
+----------------+----------+--------------+--------------------------------+
| test           | Accepted | 3            | destination type: Upstream     |
|                |          |              | namespace: gloo-system         |
|                |          |              | name: default-kubernetes-443   |
|                |          |              | weight: 1   % total: 0.33      |
|                |          |              |                                |
|                |          |              |                                |
|                |          |              | destination type: Upstream     |
|                |          |              | namespace: gloo-system         |
|                |          |              | name: extauth                  |
|                |          |              | weight: 2   % total: 0.67      |
|                |          |              | ---                            |
| test2          | Accepted | 6            | destination type: Upstream     |
|                |          |              | namespace: gloo-system         |
|                |          |              | name: gloo-system-extauth-8080 |
|                |          |              | weight: 1   % total: 0.17      |
|                |          |              |                                |
|                |          |              |                                |
|                |          |              | destination type: Upstream     |
|                |          |              | namespace: gloo-system         |
|                |          |              | name:                          |
|                |          |              | gloo-system-gateway-proxy-443  |
|                |          |              | weight: 2   % total: 0.33      |
|                |          |              |                                |
|                |          |              |                                |
|                |          |              | destination type: Upstream     |
|                |          |              | namespace: gloo-system         |
|                |          |              | name:                          |
|                |          |              | gloo-system-gateway-proxy-80   |
|                |          |              | weight: 3   % total: 0.50      |
+----------------+----------+--------------+--------------------------------+
```

File: projects/gloo/cli/pkg/cmd/create/upstreamgroup.go (hunk 1 above)
Comment (kdorosh) [reply projects/gloo/cli/pkg/cmd/create/upstreamgroup.go]: added in https://github.com/solo-io/gloo/pull/893/commits/d4ffd60cb4302b3fe81cb5ca6945d72e4fd41aa2

Comment (kdorosh) [reply projects/gloo/cli/pkg/cmd/create/upstreamgroup.go]: added for now, but for reference: https://github.com/solo-io/gloo/issues/904

TLDR; i think we should remove the functionality that was added in that PR, and redo it better
